### ISSUE TITLE
RAPTOR clustering uses KMeans and discards document order; replace with an order-aware split

### ISSUE THREAD

**Issue by mrveiss** [mrveiss/AutoBot-AI#14972]
## Problem

RAPTOR clustering uses KMeans over chunk embeddings, which discards the fact that chunks
arrive in document order. Chunk 3 and chunk 400 can land in the same cluster, and the
summary built from them is an artefact of embedding-space proximity rather than a
section of a document.

## Evidence

`autobot-backend/knowledge/pipeline/cognifiers/summarizer.py`, `_cluster_embeddings`:

```
from sklearn.cluster import KMeans
n_clusters = min(n_clusters, len(embeddings))
if n_clusters <= 1:
    return np.zeros(len(embeddings), dtype=int)
kmeans = KMeans(n_clusters=n_clusters, random_state=42, n_init=10)
return kmeans.fit_predict(embeddings)
```

`_compute_n_clusters` derives `k` from the average of `cluster_size_range` — so `k` is a
density guess, not a structural one. Ordering information is available and unused:
`ProcessedChunk` (`knowledge/pipeline/models/chunk.py`) carries `chunk_index`,
`start_offset` and `end_offset`.

Cost is also higher than it needs to be: `n_init=10` means ten full KMeans runs per
level per document.

## Proposed approach

Replace with an order-aware split:

- Compare only *consecutive* chunk embeddings (L2-normalise, cosine between neighbours)
  and split where adjacent similarity falls below a threshold derived as a percentile of
  the observed adjacent-similarity distribution. This is linear in chunk count, needs no
  `random_state` to be deterministic, and produces contiguous spans — which is what a
  "section" actually is.
- Port two guardrails with it: a cap on cluster count as a fraction of chunk count, with
  the threshold re-lowered from the sorted distribution if the cap is exceeded; and a
  small-layer collapse, where a layer at or below a small node count is summarised into a
  single parent rather than clustered.

## Risks / constraints

- **The adjacency assumption is only valid when chunks arrive in document order.** That
  holds for the current ECL pipeline, but it becomes a silent correctness bug the day a
  caller feeds unordered chunks. This must ship with an explicit `chunk_index`
  monotonicity check that raises — a comment is not sufficient.
- Percentile thresholds behave poorly at very small chunk counts; the small-layer
  collapse guardrail covers this and is not optional.
- Sequence after the tree has a retrieval sink (#14969). Optimising a clusterer whose output is
  currently discarded is wasted motion.

## Acceptance criteria

- [ ] Clusters are contiguous spans of `chunk_index`
- [ ] Unordered input raises rather than silently producing wrong clusters
- [ ] Cluster count respects the ratio cap
- [ ] A layer at or below the collapse threshold produces one parent
- [ ] Identical input produces identical output across runs with no seed pinning

---
Part of #14967

**Blocked by:** #14969 (optimising a clusterer whose output is discarded is wasted motion)
**Blocks:** Nothing.

Implementation order for the whole umbrella is in #14967.
